### PR TITLE
[#1108] feat(server): Add labels with disk path for local storage total_localfile_write_data metrics.

### PR DESCRIPTION
**Pull request by zhuyaogai**
### What changes were proposed in this pull request?

Add labels with disk path for local storage total_localfile_write_data metrics.

### Why are the changes needed?

Fix: https://github.com/apache/incubator-uniffle/issues/1108

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Add more unit test case.

### CODE REVIEW

**Comment by zhuyaogai** [comment]
@zuston BTW, do we need to add an aggregate label like `LOCAL_DISK_PATH_LABEL_ALL`?

File: server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java [hunk 1]
```diff
@@ -237,7 +237,9 @@ public Storage selectStorage(ShuffleDataReadEvent event) {
   @Override
   public void updateWriteMetrics(ShuffleDataFlushEvent event, long writeTime) {
     super.updateWriteMetrics(event, writeTime);
-    ShuffleServerMetrics.counterTotalLocalFileWriteDataSize.inc(event.getSize());
+    ShuffleServerMetrics.counterTotalLocalFileWriteDataSize
+        .labels(event.getUnderStorage().getStoragePath())
```
Comment (zuston) [inline server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java:241]: I think we should check wheater the underStorage is null?

**Review by zuston — COMMENTED** [review]
Overall lgtm. 

> @zuston BTW, do we need to add an aggregate label like LOCAL_DISK_PATH_LABEL_ALL?

I think the name of `all` is enough, like #1107 did

File: server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java (hunk 1 above)
Comment (zhuyaogai) [reply server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java]: done.

**Review by zuston — APPROVED** [review]

**Pull request merged**